public/js/AssetManager.js

```diff
@@ -417,10 +417,21 @@ function AssetManager() {
 
 				obj.material = new THREE.MeshLambertMaterial({
 					map: obj.material.map,
-					side: THREE.FrontSide,
-					skinning: true
+					side: obj.material.side,
+					skinning: obj.material.skinning
 				});
 
+				// fix self-shadows on double-sided materials
+
+				obj.material.onBeforeCompile = function(stuff) {
+					var chunk = THREE.ShaderChunk.shadowmap_pars_fragment
+						.split ('z += shadowBias')
+						.join ('z += shadowBias - 0.001');
+					stuff.fragmentShader = stuff.fragmentShader
+						.split ('#include <shadowmap_pars_fragment>')
+						.join (chunk);
+				};
+
```

Comment:
just do `console.log(stuff)` and you will see. the shader has all these #include-s before it is compiled, that 3js will replace with corresponding shader chunks. so here we have an opportunity to put the modified chunk in the shader instead, at the same time removing its #include marker so that 3js does not create a duplicate code.

Reply:
oh and the whole idea is to have material-specific shadow bias, so that characters are fixed but other parts of level are not affected.